### ISSUE TITLE
Losing type in nested RepeaterMatrixFields

### ISSUE THREAD

**Issue by aagd** [Toutouwai/RestrictRepeaterMatrix#2]
I have a _RepeaterMatrixField_ with four types, and one of those contains another RepeaterMatrixField. When I set the limit for the outer _RepeaterMatrixField_ to 1 for each type. The inner _RepeaterMatrixField_ items lose their type, when I just touch them (uncollapse) and save afterwards (no edits). I can't set back the type (from `None/undefined`) until I deactivate the limits on the outer _RepeaterMatrixField_ types. Seems like some code is not working for nested _RepeaterMatrixFields_.

All settings were applied by your hook example code.

```php
$wire->addHookAfter('RestrictRepeaterMatrix::checkRestrictions', function(HookEvent $event) {
    $field = $event->arguments('field');
    $type_restrictions = $event->return;



    if($field->name === 'outer' && !$this->user->isSuperuser()) {
        foreach($type_restrictions as $key => $value) {
          $type_restrictions[$key]['nosettings'] = true;
          $type_restrictions[$key]['nodrag'] = true;
          $type_restrictions[$key]['noclone'] = true;
          $type_restrictions[$key]['notoggle'] = true;
          $type_restrictions[$key]['notrash'] = true;

          // when the below limits are active the problem occurs

          // $type_restrictions['home_one']['limit'] = 1;
          // $type_restrictions['home_two']['limit'] = 1;
          // $type_restrictions['home_three']['limit'] = 1;
          // $type_restrictions['home_four']['limit'] = 1;
        }
    }

    if($field->name === 'inner' && !$this->user->isSuperuser()) {
        foreach($type_restrictions as $key => $value) {
          $type_restrictions[$key]['nosettings'] = true;
          $type_restrictions[$key]['nodrag'] = true;
          $type_restrictions[$key]['noclone'] = true;
          $type_restrictions[$key]['notoggle'] = true;
          $type_restrictions[$key]['notrash'] = true;
          $type_restrictions['item_large']['limit'] = 2;
          $type_restrictions['item_small']['limit'] = 2;
        }
    }
    $event->return = $type_restrictions;
});

```

**Comment by Toutouwai** [comment]
Thanks for the report. Should be fixed now in v0.1.5

**Comment by aagd** [comment]
Looks good now, many thanks for the quick response.
